broker-core/src/main/java/io/zeebe/broker/workflow/deployment/distribute/processor/state/DeploymentsStateController.java

```diff
@@ -21,98 +21,102 @@
       final long key, final PendingDeploymentDistribution pendingDeploymentDistribution) {
-    ensureIsOpened("putPendingDeployment");
 
     final int length = pendingDeploymentDistribution.getLength();
-    final byte[] bytes = new byte[length];
-    buffer.wrap(bytes);
-    pendingDeploymentDistribution.write(buffer, 0);
+    pendingDeploymentDistribution.write(valueBuffer, 0);
 
-    put(key, bytes);
+    put(pendingDeploymentColumnFamily, key, valueBuffer.byteArray(), 0, length);
   }
 
   private PendingDeploymentDistribution getPending(final long key) {
-    final byte[] bytes = get(key);
-    PendingDeploymentDistribution pending = null;
-    if (bytes != null) {
-      buffer.wrap(bytes);
-      pendingDeploymentDistribution.wrap(buffer, 0, bytes.length);
-      pending = pendingDeploymentDistribution;
-    }
-
-    return pending;
+    setLong(key);
+    final boolean successfulRead =
+        persistenceHelper.readInto(
+            pendingDeploymentDistribution,
+            pendingDeploymentColumnFamily,
+            dbLongBuffer.byteArray(),
+            0,
+            Long.BYTES);
+
+    return successfulRead ? pendingDeploymentDistribution : null;
   }
 
   public PendingDeploymentDistribution getPendingDeployment(final long key) {
-    ensureIsOpened("getPendingDeployment");
     return getPending(key);
   }
 
   public PendingDeploymentDistribution removePendingDeployment(final long key) {
-    ensureIsOpened("removePendingDeployment");
-
     final PendingDeploymentDistribution pending = getPending(key);
     if (pending != null) {
       delete(key);
     }
-
     return pending;
   }
 
   public void foreachPending(final BiConsumer<Long, PendingDeploymentDistribution> consumer) {
-    ensureIsOpened("foreachPending");
-
-    try (RocksIterator rocksIterator = getDb().newIterator()) {
-      rocksIterator.seekToFirst();
-
-      final UnsafeBuffer readBuffer = new UnsafeBuffer();
-      while (rocksIterator.isValid()) {
-
-        final byte[] keyBytes = rocksIterator.key();
-        readBuffer.wrap(keyBytes);
-
-        if (keyBytes.length == Long.BYTES) {
-          final long longKey = readBuffer.getLong(0, STATE_BYTE_ORDER);
-
-          final byte[] valueBytes = rocksIterator.value();
-          readBuffer.wrap(valueBytes);
-
-          try {
-            pendingDeploymentDistribution.wrap(readBuffer, 0, valueBytes.length);
-
-            consumer.accept(longKey, pendingDeploymentDistribution);
-          } catch (final Exception ex) {
-            // ignore
-          }
-        }
-        rocksIterator.next();
-      }
-    }
+    db.forEach(
+        pendingDeploymentColumnFamily,
+        (zbRocksEntry, iteratorControl) -> {
+          final DirectBuffer value = zbRocksEntry.getValue();
+          pendingDeploymentDistribution.wrap(value, 0, value.capacity());
+          consumer.accept(
+              zbRocksEntry.getKey().getLong(0, STATE_BYTE_ORDER), pendingDeploymentDistribution);
```

Comment:
You could use a `BiLongConsumer` to avoid that it box the key into a `Long` object.

Reply:
Or this `ObjLongConsumer`.

Reply:
Will do in next pr thanks :)